### DISCUSSION TITLE
Ideas!

### DISCUSSION THREAD

**Discussion by althonich** [losnoco/Cog#252] [Ideas]
Hello! I was wondering if any of the developers were thinking about adding playback for other sound formats such as .webm, as far as I've noticed, Cog only plays .mp3 files.

**Comment by kode54** [comment]
WebM is supported as audio-only, and only because I enabled it in FFmpeg, so you get .weba extension for now, and no chapter support. Probably not gapless, either, since gaplessness is implemented using chapters, usually.

**Comment by now-its-dark** [comment]
Hi there! Since the topic title is general, I will add an idea / question here too.

I've been curious as to how feasible it would be for Cog to support playback to system MIDI device ports, via the same dialog as the software synths. It could be useful both for software instruments which don't function as plugins, such as MUNT (MT-32 emulation) which creates a virtual midi port, or for playback to external hardware.

Maybe there is already a method or workaround to achieve this? I tried choosing the [Element](https://kushview.net/element/) AU plugin as the playback device, since it would be possible to route it from there, but it crashed Cog immediately when attempting to play a file.

**Comment by kode54** [comment]
Audio Units must generate audio, and work for offline rendering. Audio input decoders all generate audio, and not necessarily in real time.

**Comment by pgfit76** [comment]
Have you ever considered to use some more Core Audio's Audio Units in Cog?
More specifically I'm thinking about equalizers, like the parametric equalizer AUParametricEQ. A parametric EQ would be particularly useful when listening through headphones because there already exist parametric EQ curves for dozen of headphone models.

**Comment by richard-bridgeman** [comment]
This app is brilliant, I love the simplicity of the design and the focus of using folders for music organistation, which is how I group my 70gb of mp3 files.  One thing It lacks (unless its there and i cant find it) is the ability to update the mp3 tags - is this a feature that might be added at all in the future?

I'm tempted to try and add it myself.  I've done lots of coding before but never with xCode, so it might take me some time.
